### PR TITLE
Genie 1609 qodo pr agent

### PR DESCRIPTION
**Pull request by sfiresht**
<h3>Review Summary by Qodo</h3>

Configure CodeRabbit and Qodo for automated PR reviews

<code>✨ Enhancement</code> <code>⚙️ Configuration changes</code>

Grey Divider

<h3>Walkthroughs</h3>

<details open>
<summary>Description</summary>

<br/>

<pre>
• Comprehensive CodeRabbit configuration with architectural rules and security guidelines
• Added Qodo PR agent configuration for testing and security reviews
• Implemented tool-specific linting and analysis settings
• Split review responsibilities between CodeRabbit (architecture) and Qodo (testing/security)
</pre>
</details>

<details>
<summary>Diagram</summary>

<br/>

```mermaid
flowchart LR
  PR["Pull Request"]
  CR["CodeRabbit<br/>Architecture Review"]
  QD["Qodo<br/>Testing & Security"]
  PR -- "auto_review" --> CR
  PR -- "auto_review" --> QD
  CR -- "architectural rules<br/>code quality" --> Feedback1["Feedback"]
  QD -- "test coverage<br/>security checks" --> Feedback2["Feedback"]

```

</details>

Grey Divider

<h3>File Changes</h3>

<details>
<summary>1. .coderabbit.yaml
 <code>⚙️ Configuration changes</code> <code> +203/-23 </code>  </summary>  

<br/>

>Enhanced CodeRabbit with architectural rules and tools
><pre>
>• Added schema reference and language configuration
>• Restructured review settings with assertive profile and enhanced options
>• Expanded instructions with strict architectural rules for hexagonal architecture
>• Added path-specific instructions for multi-agent, backend, and ui layers
>• Configured 8 linting tools (ruff, shellcheck, yamllint, hadolint, checkov, actionlint, gitleaks,
>  trufflehog, markdownlint)
>• Disabled 30+ irrelevant tools for non-applicable tech stacks
>• Enabled chat auto-reply and knowledge base learning
></pre>
>
><a href="https://github.com/redhat-community-ai-tools/UnifAI/pull/178/files#diff-c0f93b6ec1bb8a9771fe87219aa22843985ba0cd27a23689b7be274bfb7baf57"> .coderabbit.yaml
 </a>
<hr/>

</details>

<details>
<summary>2. .pr_agent.toml
 <code>⚙️ Configuration changes</code> <code> +73/-0 </code>  </summary>  

<br/>

>New Qodo configuration for testing and security
><pre>
>• Created new Qodo PR agent configuration file
>• Configured auto-approval for PRs with no suggestions
>• Enabled auto best practices and self-review requirements
>• Defined testing standards requiring independent tests, pytest fixtures, and edge case coverage
>• Added security guidelines for secrets, input sanitization, and authorization checks
>• Specified hard compliance rules including type hints, no TODOs, and no bare except blocks
>• Configured inline comments with severity threshold and PR description generation
></pre>
>
><a href="https://github.com/redhat-community-ai-tools/UnifAI/pull/178/files#diff-356a4c0b1558da9e4be849aa64f19af78488ec6819f379e21ae93c53e750fbe7"> .pr_agent.toml
 </a>
<hr/>

</details>

Grey Divider

<a href="https://www.qodo.ai">Qodo Logo</a>

### CODE REVIEW

**Comment by sfiresht** [comment]
/agentic_review --review_agent.demand_self_review=true --review_agent.approve_pr_on_self_review=true --review_agent.issues_user_guidelines="This repository represents an agentic AI system with a modular, layered architecture. Each of the main directories (multi-agent/, backend/, ui/) contains README.md and ARCHITECTURE.md files that define architectural intent, responsibilities, and boundaries. These documents should be treated as the source of truth. During reviews, always cross-check code changes against the architectural guidance in those files. FOCUS AREAS: architecture, maintainability, modularity, separation of concerns, dependency boundaries, layering, testability. Please pay special attention to: whether new components are placed in the correct layer and folder, whether responsibilities align with the architecture defined in the directory-level README.md and ARCHITECTURE.md files, whether any architectural boundaries are violated, whether new dependencies increase coupling in undesirable ways, whether changes weaken modularity or future extensibility, whether agent backend and UI concerns are leaking across layers. Ensure that all new error-handling paths contain appropriate logging statements. Prefer architectural and system-level feedback over stylistic feedback. When relevant, explicitly reference architectural mismatches. Act as a senior software engineer deeply familiar with this project's architecture."

File: .github/workflows/cr-assistant-qodo.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      env:
+        # OPENAI_KEY: ${{ secrets.OPENAI_KEY }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        CONFIG__MODEL: "gemini/gemini-3.1-flash-lite-preview"
+        CONFIG__FALLBACK_MODELS: '["gemini/gemini-2.5-pro"]'
+        GOOGLE_AI_STUDIO__GEMINI_API_KEY: ${{ secrets.GEMINI_API_KEY }}
+        github_action_config.auto_review: "true"
+        github_action_config.auto_describe: "true"
+        github_action_config.auto_improve: "true"
+        github_action_config.pr_actions: '["opened", "reopened", "ready_for_review", "synchronize"]'
```
[the remark this answers is not in the document]
Comment (sfiresht) [reply .github/workflows/cr-assistant-qodo.yaml]: removed the file

File: .github/workflows/cr-assistant-qodo.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        description: 'Branch to scan'
+        required: true
+        default: 'main'
+  # push:
+  #   branches:
+  #       - GENIE-1609_qodo_pr_agent
+  # pull_request:
+  #   types: [opened, synchronize, reopened, ready_for_review, converted_to_draft ]
+  #   # branches:
+    #     - GENIE-1609_qodo_pr_agent
```
[the remark this answers is not in the document]
Comment (sfiresht) [reply .github/workflows/cr-assistant-qodo.yaml]: removed the file

File: .github/workflows/cr-assistant-qodo.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+jobs:
+  pr_agent_job:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == true
```
[the remark this answers is not in the document]
Comment (sfiresht) [reply .github/workflows/cr-assistant-qodo.yaml]: removed the file

**Comment by sfiresht** [comment]
/agentic_review --review_agent.issues_user_guidelines="Focus on architecture and dependency boundaries."

**Comment by sfiresht** [comment]
/agentic_review --review_agent.issues_user_guidelines="Focus on architecture and dependency boundaries."

**Comment by sfiresht** [comment]
/agentic_review --review_agent.issues_user_guidelines="Focus on architecture and dependency boundaries."

**Comment by sfiresht** [comment]
/agentic_review --review_agent.comments_location_policy="inline" --review_agent.inline_comments_severity_threshold=1

**Comment by sfiresht** [comment]
/agentic_review --config.enable_auto_approval=true --config.auto_approve_for_no_suggestions=true --review_agent.comments_location_policy="inline" --review_agent.inline_comments_severity_threshold=1

**Comment by sfiresht** [comment]
/agentic_review --config.enable_auto_approval=true --config.auto_approve_for_no_suggestions=true --review_agent.comments_location_policy="inline" --review_agent.inline_comments_severity_threshold=1

**Review by Lina-AbuYousef — DISMISSED** [review]

**Review by Nirsisr — APPROVED** [review]

**Review by shanitzvii — APPROVED** [review]
